lib/LaTeXML/Package/ifplatform.sty.ltxml

```diff
@@ -0,0 +1,68 @@
+# -*- mode: Perl -*-
+# /=====================================================================\ #
+# |  ifplatform                                                         | #
+# | Implementation for LaTeXML                                          | #
+# |=====================================================================| #
+# | Part of LaTeXML:                                                    | #
+# |  Public domain software, produced as part of work done by the       | #
+# |  United States Government & not subject to copyright in the US.     | #
+# |=====================================================================| #
+# | Thanks to Tom Wiesing <tom.wiesing@gmail.com>.                      | #
+# | Additional thanks to GitHub user "aivokalu" for part of the         | #
+# | original implementation.                                            | #
+# |---------------------------------------------------------------------| #
+# | Bruce Miller <bruce.miller@nist.gov>                        #_#     | #
+# | http://dlmf.nist.gov/LaTeXML/                              (o o)    | #
+# \=========================================================ooo==U==ooo=/ #
+package LaTeXML::Package::Pool;
+use strict;
+use warnings;
+use LaTeXML::Package;
+
+# Define the names for the common operating systems
+DefMacro('\windowsname',    'Windows');
+DefMacro('\notwindowsname', '*NIX');
+DefMacro('\linuxname',      'Linux');
+DefMacro('\macosxname',     'Mac\,OS\,X');
+DefMacro('\cygwinname',     'Cygwin');
+
+# internal 'if' being used to detect if shell escape is enabled.
+# The concept doesn't exist in LaTeXML the way it exists in pdflatex,
+# so we hardcode it as true.
+DefConditional('\ifshellescape', sub { 1; });
```

Comment:
Not so much that  the concept doesn't exist, but that we haven't implemented it, and security is a big disincentive to do so.  Wouldn't it be better to return 0, so that code won't then *try* to execute a shell escape and have it ignored (fail)?

Reply:
The reason i have it set to true is that the package internally uses `--shell-escape` to determine if Mac, Linux or other OSs are being used. From the manual:
> If you also wish to determine the difference between which Unix-variant you are using (i.e., also detect \iflinux, \ifmacosx, and \ifcygwin) then shell escape must be enabled. This is achieved by using the -shell-escape command line option when executing LATEX
When shell escape is not enabled, these always return false. In the binding we can of course detect those operating systems without needing to enable (or even implement) such an option. 

I have a strong suspicion (although no data to back this up) that `\ifshellescape` is also misused outside the package to check if non-windows OS detection is enabled. And since it is enabled in the binding, in my view it makes sense to set it to 1.

Reply:
OK, let's go with that, but keep in the back of our mind that it's a bit weird :>